crash in mapbox

@cyBerta got the following crash by just opening the map using the map_filtering_2 branch. device is my nexus4 with android8.1/lineage

```
2019-03-27 10:27:56.096 1587-1587/? A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x18 in tid 1587 (.messenger.beta), pid 1587 (.messenger.beta)
2019-03-27 10:27:56.233 1912-1912/? I/crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
2019-03-27 10:27:56.235 296-296/? I//system/bin/tombstoned: received crash request for pid 1587
2019-03-27 10:27:56.241 1912-1912/? I/crash_dump32: performing dump of process 1587 (target tid = 1587)
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: LineageOS Version: '15.1-20190207-NIGHTLY-mako'
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: Build fingerprint: 'google/occam/mako:5.1.1/LMY48T/2237560:user/release-keys'
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: Revision: '0'
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: ABI: 'arm'
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: pid: 1587, tid: 1587, name: .messenger.beta  >>> com.b44t.messenger.beta <<<
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x18
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG: Cause: null pointer dereference
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG:     r0 beee2af0  r1 00000004  r2 00000003  r3 00000000
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG:     r4 beee2ae8  r5 999d0478  r6 94be7380  r7 beee2b20
2019-03-27 10:27:56.245 1912-1912/? A/DEBUG:     r8 beee2b38  r9 999d0498  sl 00000000  fp b2605000
2019-03-27 10:27:56.246 1912-1912/? A/DEBUG:     ip b55dbd58  sp beee2ad8  lr 98213043  pc 98213042  cpsr 20070030
2019-03-27 10:27:56.307 1912-1912/? A/DEBUG: backtrace:
2019-03-27 10:27:56.307 1912-1912/? A/DEBUG:     #00 pc 00147042  /data/app/com.b44t.messenger.beta-bPNOSxJoRDwv4eWusGhFxw==/lib/arm/libmapbox-gl.so
2019-03-27 10:27:56.307 1912-1912/? A/DEBUG:     #01 pc 0014a045  /data/app/com.b44t.messenger.beta-bPNOSxJoRDwv4eWusGhFxw==/lib/arm/libmapbox-gl.so
2019-03-27 10:27:56.307 1912-1912/? A/DEBUG:     #02 pc 0040ca79  /system/lib/libart.so (art_quick_generic_jni_trampoline+40)
2019-03-27 10:27:56.307 1912-1912/? A/DEBUG:     #03 pc 00408575  /system/lib/libart.so (art_quick_invoke_stub_internal+68)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #04 pc 0040d6e5  /system/lib/libart.so (art_quick_invoke_stub+228)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #05 pc 000b00eb  /system/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+138)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #06 pc 002031e7  /system/lib/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+222)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #07 pc 001fe735  /system/lib/libart.so (_ZN3art11interpreter6DoCallILb0ELb0EEEbPNS_9ArtMethodEPNS_6ThreadERNS_11ShadowFrameEPKNS_11InstructionEtPNS_6JValueE+588)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #08 pc 003f359b  /system/lib/libart.so (MterpInvokeDirect+254)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #09 pc 003faa14  /system/lib/libart.so (ExecuteMterpImpl+14484)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #10 pc 001e55c9  /system/lib/libart.so (art::interpreter::Execute(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame&, art::JValue, bool)+340)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #11 pc 001e9d77  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame*, art::JValue*)+142)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #12 pc 001fe71d  /system/lib/libart.so (_ZN3art11interpreter6DoCallILb0ELb0EEEbPNS_9ArtMethodEPNS_6ThreadERNS_11ShadowFrameEPKNS_11InstructionEtPNS_6JValueE+564)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #13 pc 003f26fb  /system/lib/libart.so (MterpInvokeVirtual+482)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #14 pc 003fa914  /system/lib/libart.so (ExecuteMterpImpl+14228)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #15 pc 001e55c9  /system/lib/libart.so (art::interpreter::Execute(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame&, art::JValue, bool)+340)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #16 pc 001e9d77  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame*, art::JValue*)+142)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #17 pc 001fe71d  /system/lib/libart.so (_ZN3art11interpreter6DoCallILb0ELb0EEEbPNS_9ArtMethodEPNS_6ThreadERNS_11ShadowFrameEPKNS_11InstructionEtPNS_6JValueE+564)
2019-03-27 10:27:56.308 1912-1912/? A/DEBUG:     #18 pc 003f32dd  /system/lib/libart.so (MterpInvokeInterface+1080)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #19 pc 003fab14  /system/lib/libart.so (ExecuteMterpImpl+14740)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #20 pc 001e55c9  /system/lib/libart.so (art::interpreter::Execute(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame&, art::JValue, bool)+340)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #21 pc 001e9d77  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame*, art::JValue*)+142)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #22 pc 001fe71d  /system/lib/libart.so (_ZN3art11interpreter6DoCallILb0ELb0EEEbPNS_9ArtMethodEPNS_6ThreadERNS_11ShadowFrameEPKNS_11InstructionEtPNS_6JValueE+564)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #23 pc 003f26fb  /system/lib/libart.so (MterpInvokeVirtual+482)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #24 pc 003fa914  /system/lib/libart.so (ExecuteMterpImpl+14228)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #25 pc 001e55c9  /system/lib/libart.so (art::interpreter::Execute(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame&, art::JValue, bool)+340)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #26 pc 001e9d77  /system/lib/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame*, art::JValue*)+142)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #27 pc 001fe71d  /system/lib/libart.so (_ZN3art11interpreter6DoCallILb0ELb0EEEbPNS_9ArtMethodEPNS_6ThreadERNS_11ShadowFrameEPKNS_11InstructionEtPNS_6JValueE+564)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #28 pc 003f26fb  /system/lib/libart.so (MterpInvokeVirtual+482)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #29 pc 003fa914  /system/lib/libart.so (ExecuteMterpImpl+14228)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #30 pc 001e55c9  /system/lib/libart.so (art::interpreter::Execute(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame&, art::JValue, bool)+340)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #31 pc 001e9cc5  /system/lib/libart.so (art::interpreter::EnterInterpreterFromEntryPoint(art::Thread*, art::DexFile::CodeItem const*, art::ShadowFrame*)+92)
2019-03-27 10:27:56.309 1912-1912/? A/DEBUG:     #32 pc 003e7ee7  /system/lib/libart.so (artQuickToInterpreterBridge+934)
2019-03-27 10:27:56.310 1912-1912/? A/DEBUG:     #33 pc 0040caf1  /system/lib/libart.so (art_quick_to_interpreter_bridge+32)
2019-03-27 10:27:56.310 1912-1912/? A/DEBUG:     #34 pc 000b10c1  /dev/ashmem/dalvik-jit-code-cache (deleted)
2019-03-27 10:28:02.076 296-296/? W//system/bin/tombstoned: crash socket received short read of length 0 (expected 12)
2019-03-27 10:28:02.079 418-440/system_process I/BootReceiver: Copying /data/tombstones/tombstone_02 to DropBox (SYSTEM_TOMBSTONE)
2019-03-27 10:28:02.150 418-436/system_process W/zygote: kill(-1587, 9) failed: No such process
2019-03-27 10:28:02.151 418-771/system_process I/ActivityManager: Process com.b44t.messenger.beta (pid 1587) has died: fore TOP 
```

one problem might be that the stuff has to be executed in the ui-thread, 
eg. the used EmojiTextView crashes with a failed asset at https://github.com/deltachat/deltachat-android/blob/91be337fb68cb75712c64a3d01b94409329af675/src/org/thoughtcrime/securesms/components/emoji/parsing/EmojiPageBitmap.java#L39 :

```
2019-03-27 11:35:45.859 3155-3191/com.b44t.messenger.beta E/AndroidRuntime: FATAL EXCEPTION: org.thoughtcrime.securesms.map.GenerateInfoWindowTask
    Process: com.b44t.messenger.beta, PID: 3155
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:353)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.AssertionError: Main-thread assertion failed.
        at org.thoughtcrime.securesms.util.Util.assertMainThread(Util.java:263)
        at org.thoughtcrime.securesms.components.emoji.parsing.EmojiPageBitmap.get(EmojiPageBitmap.java:39)
        at org.thoughtcrime.securesms.components.emoji.EmojiProvider.getEmojiDrawable(EmojiProvider.java:116)
        at org.thoughtcrime.securesms.components.emoji.EmojiProvider.emojify(EmojiProvider.java:94)
        at org.thoughtcrime.securesms.components.emoji.EmojiTextView.setText(EmojiTextView.java:83)
        at android.widget.TextView.setText(TextView.java:5207)
        at org.thoughtcrime.securesms.map.GenerateInfoWindowTask.doInBackground(GenerateInfoWindowTask.java:111)
        at org.thoughtcrime.securesms.map.GenerateInfoWindowTask.doInBackground(GenerateInfoWindowTask.java:50)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 
```

--- a/res/layout/map_bubble_layout.xml
+++ b/res/layout/map_bubble_layout.xml
@@ -48,6 +48,7 @@
             android:paddingTop="@dimen/message_bubble_top_padding"
             android:textColor="?conversation_item_outgoing_text_primary_color"
             android:textColorLink="?conversation_item_outgoing_text_primary_color"
+            app:createInBackground="true"
             app:scaleEmojis="true"
             tools:text="Mango pickle lorem ipsum"/>
     </LinearLayout>

--- a/res/values/attrs.xml
+++ b/res/values/attrs.xml
@@ -201,6 +201,7 @@
 
     <declare-styleable name="EmojiTextView">
         <attr name="scaleEmojis" format="boolean" />
+        <attr name="createInBackground" format="boolean" />
     </declare-styleable>
 
     <declare-styleable name="RingtonePreference">

--- a/src/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java
+++ b/src/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java
@@ -11,7 +11,6 @@ import android.graphics.Rect;
 import android.graphics.drawable.Drawable;
 import android.os.Build.VERSION;
 import android.os.Build.VERSION_CODES;
-import android.support.annotation.DrawableRes;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.text.Spannable;
@@ -25,8 +24,8 @@ import org.thoughtcrime.securesms.components.emoji.parsing.EmojiPageBitmap;
 import org.thoughtcrime.securesms.components.emoji.parsing.EmojiParser;
 import org.thoughtcrime.securesms.components.emoji.parsing.EmojiTree;
 import org.thoughtcrime.securesms.util.FutureTaskListener;
-import org.thoughtcrime.securesms.util.Util;
 import org.thoughtcrime.securesms.util.Pair;
+import org.thoughtcrime.securesms.util.Util;
 
 import java.io.IOException;
 import java.util.concurrent.ExecutionException;

--- a/src/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java
+++ b/src/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java
@@ -11,6 +11,7 @@ import android.graphics.Rect;
 import android.graphics.drawable.Drawable;
 import android.os.Build.VERSION;
 import android.os.Build.VERSION_CODES;
+import android.support.annotation.DrawableRes;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.text.Spannable;
@@ -27,12 +28,13 @@ import org.thoughtcrime.securesms.util.FutureTaskListener;
 import org.thoughtcrime.securesms.util.Util;
 import org.thoughtcrime.securesms.util.Pair;
 
+import java.io.IOException;
 import java.util.concurrent.ExecutionException;
 
 class EmojiProvider {
 
   private static final    String        TAG      = EmojiProvider.class.getSimpleName();
-  private static volatile EmojiProvider instance = null;
+  protected static volatile EmojiProvider instance = null;
   private static final    Paint         paint    = new Paint(Paint.FILTER_BITMAP_FLAG | Paint.ANTI_ALIAS_FLAG);
 
   private final EmojiTree emojiTree = new EmojiTree();
@@ -56,7 +58,7 @@ class EmojiProvider {
     return instance;
   }
 
-  private EmojiProvider(Context context) {
+  protected EmojiProvider(Context context) {
     this.decodeScale = Math.min(1f, context.getResources().getDimension(R.dimen.emoji_drawer_size) / EMOJI_RAW_HEIGHT);
     this.verticalPad = EMOJI_VERT_PAD * this.decodeScale;
 
@@ -81,17 +83,18 @@ class EmojiProvider {
   }
 
   @Nullable Spannable emojify(@Nullable CharSequence text, @NonNull TextView tv) {
-    return emojify(getCandidates(text), text, tv);
+    return emojify(getCandidates(text), text, tv, false);
   }
 
   @Nullable Spannable emojify(@Nullable EmojiParser.CandidateList matches,
                               @Nullable CharSequence text,
-                              @NonNull TextView tv) {
+                              @NonNull TextView tv,
+                              boolean background) {
     if (matches == null || text == null) return null;
     SpannableStringBuilder      builder = new SpannableStringBuilder(text);
 
     for (EmojiParser.Candidate candidate : matches) {
-      Drawable drawable = getEmojiDrawable(candidate.getDrawInfo());
+      Drawable drawable = getEmojiDrawable(candidate.getDrawInfo(), background);
 
       if (drawable != null) {
         builder.setSpan(new EmojiSpan(drawable, tv), candidate.getStartIndex(), candidate.getEndIndex(),
@@ -104,24 +107,32 @@ class EmojiProvider {
 
   @Nullable Drawable getEmojiDrawable(CharSequence emoji) {
     EmojiDrawInfo drawInfo = emojiTree.getEmoji(emoji, 0, emoji.length());
-    return getEmojiDrawable(drawInfo);
+    return getEmojiDrawable(drawInfo, false);
   }
 
-  private @Nullable Drawable getEmojiDrawable(@Nullable EmojiDrawInfo drawInfo) {
+  protected @Nullable Drawable getEmojiDrawable(@Nullable EmojiDrawInfo drawInfo, boolean background) {
     if (drawInfo == null)  {
       return null;
     }
 
     final EmojiDrawable drawable = new EmojiDrawable(drawInfo, decodeScale);
-    drawInfo.getPage().get().addListener(new FutureTaskListener<Bitmap>() {
-      @Override public void onSuccess(final Bitmap result) {
-        Util.runOnMain(() -> drawable.setBitmap(result));
+    if (background) {
+      try {
+        drawable.setBitmap(drawInfo.getPage().loadPage(), background);
+      } catch (IOException e) {
+        e.printStackTrace();
       }
+    } else {
+      drawInfo.getPage().get().addListener(new FutureTaskListener<Bitmap>() {
+        @Override public void onSuccess(final Bitmap result) {
+          Util.runOnMain(() -> drawable.setBitmap(result));
+        }
 
-      @Override public void onFailure(ExecutionException error) {
-        Log.w(TAG, error);
-      }
-    });
+        @Override public void onFailure(ExecutionException error) {
+          Log.w(TAG, error);
+        }
+      });
+    }
     return drawable;
   }
 
@@ -167,7 +178,14 @@ class EmojiProvider {
 
     @TargetApi(VERSION_CODES.HONEYCOMB_MR1)
     public void setBitmap(Bitmap bitmap) {
-      Util.assertMainThread();
+      setBitmap(bitmap, false);
+    }
+
+    @TargetApi(VERSION_CODES.HONEYCOMB_MR1)
+    public void setBitmap(Bitmap bitmap, boolean background) {
+      if (!background) {
+        Util.assertMainThread();
+      }
       if (VERSION.SDK_INT < VERSION_CODES.HONEYCOMB_MR1 || bmp == null || !bmp.sameAs(bitmap)) {
         bmp = bitmap;
         invalidateSelf();

--- a/src/org/thoughtcrime/securesms/components/emoji/EmojiTextView.java
+++ b/src/org/thoughtcrime/securesms/components/emoji/EmojiTextView.java
@@ -22,6 +22,7 @@ import org.thoughtcrime.securesms.util.Util;
 public class EmojiTextView extends AppCompatTextView {
 
   private final boolean scaleEmojis;
+  private final boolean createInBackground;
 
   private CharSequence previousText;
   private BufferType   previousBufferType;
@@ -42,6 +43,7 @@ public class EmojiTextView extends AppCompatTextView {
 
     TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.EmojiTextView, 0, 0);
     scaleEmojis = a.getBoolean(R.styleable.EmojiTextView_scaleEmojis, false);
+    createInBackground = a.getBoolean(R.styleable.EmojiTextView_createInBackground, false);
     a.recycle();
 
     a = context.obtainStyledAttributes(attrs, new int[]{android.R.attr.textSize});
@@ -80,7 +82,7 @@ public class EmojiTextView extends AppCompatTextView {
       return;
     }
 
-    CharSequence emojified = provider.emojify(candidates, text, this);
+    CharSequence emojified = provider.emojify(candidates, text, this, createInBackground);
     super.setText(emojified, BufferType.SPANNABLE);
 
     // Android fails to ellipsize spannable strings. (https://issuetracker.google.com/issues/36991688)
@@ -113,7 +115,7 @@ public class EmojiTextView extends AppCompatTextView {
             .append(ellipsized.subSequence(0, ellipsized.length()));
 
         EmojiParser.CandidateList newCandidates = EmojiProvider.getInstance(getContext()).getCandidates(newContent);
-        CharSequence              emojified     = EmojiProvider.getInstance(getContext()).emojify(newCandidates, newContent, this);
+        CharSequence              emojified     = EmojiProvider.getInstance(getContext()).emojify(newCandidates, newContent, this, createInBackground);
 
         super.setText(emojified, BufferType.SPANNABLE);
       }

--- a/src/org/thoughtcrime/securesms/components/emoji/parsing/EmojiPageBitmap.java
+++ b/src/org/thoughtcrime/securesms/components/emoji/parsing/EmojiPageBitmap.java
@@ -67,7 +67,7 @@ public class EmojiPageBitmap {
     return task;
   }
 
-  private Bitmap loadPage() throws IOException {
+  public Bitmap loadPage() throws IOException {
     if (bitmapReference != null && bitmapReference.get() != null) return bitmapReference.get();
 
     try {

--- a/src/org/thoughtcrime/securesms/map/GenerateInfoWindowTask.java
+++ b/src/org/thoughtcrime/securesms/map/GenerateInfoWindowTask.java
@@ -21,6 +21,7 @@ import com.mapbox.geojson.Feature;
 
 import org.thoughtcrime.securesms.R;
 import org.thoughtcrime.securesms.components.ConversationItemFooter;
+import org.thoughtcrime.securesms.components.emoji.EmojiTextView;
 import org.thoughtcrime.securesms.connect.DcHelper;
 import org.thoughtcrime.securesms.util.BitmapUtil;
 import org.thoughtcrime.securesms.util.DateUtils;
@@ -77,7 +78,7 @@ public class GenerateInfoWindowTask extends AsyncTask<ArrayList<Feature>, HashMa
                 LinearLayout bubbleLayout = (LinearLayout)
                         inflater.inflate(R.layout.map_bubble_layout, null);
                 bubbleLayout.setBackgroundResource(R.drawable.message_bubble_background_received_alone);
-                TextView conversationItemBody = bubbleLayout.findViewById(R.id.conversation_item_body);
+                EmojiTextView conversationItemBody = bubbleLayout.findViewById(R.id.conversation_item_body);
                 Locale locale = DynamicLanguage.getSelectedLocale(callbackRef.get().getContext());
                 int messageId = (int) feature.getNumberProperty(MESSAGE_ID);
                 int contactId = (int) feature.getNumberProperty(CONTACT_ID);

